mobile_verifier/src/coverage.rs

```diff
@@ -0,0 +1,466 @@
+            indoor,
+            signal_level,
+            coverage_claim_time,
+            cbsd_id,
+            ..
+        }) = covered_hexes.next().await.transpose()?
+        {
+            self.hexes
+                .entry(CellIndex::try_from(hex as u64).unwrap())
+                .or_default()
+                .entry(signal_level)
+                .or_default()
+                .push(CoverageLevel {
+                    cbsd_id,
+                    coverage_claim_time,
+                    indoor,
+                    signal_level,
+                    hotspot: hotspot.clone(),
+                });
+        }
+
+        Ok(())
+    }
+
+    pub fn into_iter(self) -> impl Iterator<Item = CoverageReward> {
+        self.hexes
+            .into_values()
+            .map(|mut radios| {
+                radios.pop_last().map(|(_, radios)| {
+                    radios
+                        .into_sorted_vec()
+                        .into_iter()
+                        .take(MAX_RADIOS_PER_HEX)
+                        .map(|cl| CoverageReward {
+                            points: cl.coverage_points().unwrap(),
+                            hotspot: cl.hotspot,
+                            cbsd_id: cl.cbsd_id,
+                        })
+                })
+            })
+            .flatten()
+            .flatten()
+            .filter(|r| r.points > Decimal::ZERO)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use chrono::NaiveDate;
+    use futures::stream::iter;
+
+    fn default_hex_coverage(cbsd_id: &str, signal_level: SignalLevel) -> HexCoverage {
+        HexCoverage {
+            uuid: Uuid::new_v4(),
+            hex: 0x8a1fb46622dffff_u64 as i64,
+            indoor: false,
+            cbsd_id: cbsd_id.to_string(),
+            signal_level,
+            coverage_claim_time: DateTime::<Utc>::MIN_UTC,
+        }
+    }
+
+    #[tokio::test]
+    async fn ensure_max_signal_level_selected() {
+        let owner: PublicKeyBinary = "112NqN2WWMwtK29PMzRby62fDydBJfsCLkCAf392stdok48ovNT6"
+            .parse()
+            .expect("failed owner parse");
+        let mut covered_hexes = CoveredHexes::default();
+        covered_hexes
+            .aggregate_coverage(
+                &owner,
+                iter(vec![
+                    anyhow::Ok(default_hex_coverage("1", SignalLevel::No)),
+                    anyhow::Ok(default_hex_coverage("2", SignalLevel::Low)),
+                    anyhow::Ok(default_hex_coverage("3", SignalLevel::Medium)),
+                    anyhow::Ok(default_hex_coverage("4", SignalLevel::High)),
+                    anyhow::Ok(default_hex_coverage("5", SignalLevel::Medium)),
+                    anyhow::Ok(default_hex_coverage("6", SignalLevel::Low)),
+                    anyhow::Ok(default_hex_coverage("7", SignalLevel::No)),
+                ]),
+            )
+            .await
+            .unwrap();
+        let rewards: Vec<_> = covered_hexes.into_iter().collect();
```

Comment:
Is my understanding correct that this only returns a single radio and not the top 5 as the requirement is to return up to 5 radios only of the *same* strength and as there is only a single radio with High only that radio is returned ? If 2 radios had high then 2 would have been returned here ?

Reply:
Yup, requirement is oldest five of the top strength